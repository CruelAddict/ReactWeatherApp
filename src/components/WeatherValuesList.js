import React from 'react'

export default (props) => {
    return <div className={"weather-value-list"}>
        <div className={"weather-value-row"}>
            <span className={"weather-value-key"}>Ветер</span>
            <span className={"weather-value-value"}>{props.state.weather_data.wind} m/s</span>
        </div>
        <div className={"weather-value-row"}>
            <span className={"weather-value-key"}>Облачность</span>
            <span className={"weather-value-value"}>{props.state.weather_data.cloudiness.split(' ').map((s) =>
                s.charAt(0).toUpperCase() + s.substring(1)).join(' ')}</span>
        </div>
        <div className={"weather-value-row"}>
            <span className={"weather-value-key"}>Давление</span>
            <span className={"weather-value-value"}>{props.state.weather_data.pressure} hpa</span>
        </div>
        <div className={"weather-value-row"}>
            <span className={"weather-value-key"}>Влажность</span>
            <span className={"weather-value-value"}>{props.state.weather_data.humidity} %</span>
        </div>
        <div className={"weather-value-row"}>
            <span className={"weather-value-key"}>Координаты</span>
            <span
                className={"weather-value-value"}>[{props.state.coordinates.longitude.toFixed(2)}, {props.state.coordinates.latitude.toFixed(2)}]</span>
        </div>
    </div>
}
