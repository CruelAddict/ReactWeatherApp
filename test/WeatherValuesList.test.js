import React from 'react';
import { shallow, render, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import * as Enzyme from "enzyme";
Enzyme.configure({ adapter: new Adapter() });

import WeatherValuesList from '../src/components/WeatherValuesList';


describe ('WeatherValuesList', () => {

    it('Displays weather data items for a city', () => {
        const panel = shallow(
            <WeatherValuesList state={{
                isLoading: false,
                city: "London",
                coordinates: {
                    latitude: 51.51,
                    longitude: -0.13
                },
                isFailed: false,
                weather_data: {
                    cloudiness: "overcast clouds",
                    humidity: 87,
                    icon: "blob:http://localhost:3000/983a1031-dd0b-42bb-a079-8feb35f05a10",
                    pressure: 1016,
                    temperature: 3.9700000000000273,
                    wind: 2.1
                }
            }}/>
        );
        expect(panel).toMatchSnapshot()
    });

});
