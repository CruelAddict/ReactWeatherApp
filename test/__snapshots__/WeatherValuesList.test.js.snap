// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherValuesList Displays weather data items for a city 1`] = `
<div
  className="weather-value-list"
>
  <div
    className="weather-value-row"
  >
    <span
      className="weather-value-key"
    >
      Ветер
    </span>
    <span
      className="weather-value-value"
    >
      2.1
       m/s
    </span>
  </div>
  <div
    className="weather-value-row"
  >
    <span
      className="weather-value-key"
    >
      Облачность
    </span>
    <span
      className="weather-value-value"
    >
      Overcast Clouds
    </span>
  </div>
  <div
    className="weather-value-row"
  >
    <span
      className="weather-value-key"
    >
      Давление
    </span>
    <span
      className="weather-value-value"
    >
      1016
       hpa
    </span>
  </div>
  <div
    className="weather-value-row"
  >
    <span
      className="weather-value-key"
    >
      Влажность
    </span>
    <span
      className="weather-value-value"
    >
      87
       %
    </span>
  </div>
  <div
    className="weather-value-row"
  >
    <span
      className="weather-value-key"
    >
      Координаты
    </span>
    <span
      className="weather-value-value"
    >
      [
      -0.13
      , 
      51.51
      ]
    </span>
  </div>
</div>
`;
